### PR TITLE
[chore] Upgrade seed dependencies

### PR DESCRIPTION
**Pull request by hippee-lee**
## Notes

I tried to bring our seed up close to the projects that are generated with the latest angular-cli tool (1.2.1). This included 

- running `npm outdated` and applying most of the recomended changes
- a few code changes and configuration changes for the updated deps
- Modified app code for changes in Clarity v0.10.x
- Modified code for changes to Jasmine + additional @types/jasminewd2
- Swapped out HeadlessChrome for PhantomJS

`ng start`, `ng test`, and `ng e2e` all run as expected.

Signed-off-by: Matt Hippely <mhippely@vmware.com>

### CODE REVIEW

**Comment by hippee-lee** [comment]
@jeeyun - I won't merge this in until you give approval b/c you are using it in the wild soon.

**Review by jeeyun — APPROVED** [review]

**Pull request merged**
